### PR TITLE
Reorganize SIG Release documentation

### PR DESCRIPTION
**Pull request by justaugustus**
This shuffles some of the documentation around to more explicitly delineate the Release Team as a subproject of the SIG Release and not necessarily the owners of `k/sig-release`.

No attempts have been made to refine the content of the documentation; that's happening as a series of issues assigned to the 1.11 Release Team members.

/cc @calebamiles @jdumars @jberkus @tpepper

### CODE REVIEW

**Comment by justaugustus** [comment]
Holding for comments / approvals.
/hold

File: release-team/OWNERS [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# See the OWNERS docs: https://git.k8s.io/community/contributors/guide/owners.md
```
[the remark this answers is not in the document]
Comment (justaugustus) [reply release-team/OWNERS]: It's useful, but not strictly necessary. Copied over from other OWNERS files in the repo.

[the remark this answers is not in the document]
Comment (tpepper) [reply release-team/OWNERS]: For some reason my multi-tasked brain took that to be saying somehow this OWNERS file's content was related to some other one.

It seems like other OWNERS files in other k/ repos generally don't include this at their head.  A Google search on "what is a kubernetes OWNERS file" gets relevant info, so the docs are discoverable.  I'd rather leave it out so the future has less maintenance and dangling links of there's a doc move in the contrib guide.

But not a major issue.  ¯\_(ツ)_/¯

**Comment by tpepper** [comment]
Aha and that update proves it's a maintenance issue.  The file has recently moved and one of the links you removed was a 404 already. :\

Cleaning house!

**Comment by tpepper** [comment]
should get some more folks to chime in but

/lgtm

**Comment by jdumars** [comment]
/lgtm 
Thank you!

**Comment by justaugustus** [comment]
Thanks Jaice!
/hold cancel

**Pull request merged**